base/utferror.jl

```diff
@@ -0,0 +1,30 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+##\brief      Error messages for Unicode / UTF support
```

Comment:
@timholy I think the big difference is writing for computers vs. writing for humans. If you write doxygen/javadoc/whatever syntax you end just trying to satisfy the tooling, whereas free text forces you to think about the human on the other end. On top of that, text is just so natural – there's much less cognitive overhead for both reader and writer (who may well be new to programming) when you don't have to remember how a `\param[in]` statement works every time you want help. Reduce the friction and more people get involved, simple as that.

(What do you even lose if you don't have structured data? We've managed fine without it so far, and I've never seen a truly compelling use case that would outweigh the extra complexity.)